File: Proj4/Proj4a/e3/bubblesort.c
#include <stdio.h>

void bubble_sort(int arr[], int n) {
    int i, j, temp;
    for (i = 0; i < n-1; i++) {
        for (j = 0; j < n-i-1; j++) {
            if (arr[j] > arr[j+1]) {
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }
}

int main() {
    int arr[10000];
    fgets(arr, sizeof(arr), stdin);
    for (int i = 0;i< 10000;i++)
    {
        if (arr[i] == '\n') arr[i] = '\0';
    }
    int n = sizeof(arr) / sizeof(arr[0]);
    bubble_sort(arr, n);
    for (int i = 0; i < n; i++)
        printf("%d ", arr[i]);
    printf("\n");
    return 0;
}

File: Proj4/Proj4a/e3/BubbleSort.java
public class BubbleSort {
    void bubbleSort(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n-1; i++) {
            for (int j = 0; j < n-i-1; j++) {
                if (arr[j] > arr[j+1]) {
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
    }

    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        int arr[10000];
        for (int i = 0; i < 10000; i++) {
              arr[i] = scanner.nextInt();
              if (entrada.isEmpty()) {
                  break;
              }
        }
        BubbleSort ob = new BubbleSort();
        ob.bubbleSort(arr);
        for (int i = 0; i < arr.length; ++i)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
}
